pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,137 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	// if there is a repo layer link, return its desc
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
```

Comment:
> Is there a cache for this already in the registry?

Ahh, I completely forgot. There is. It's a wrapper just above this wrapper. In our default config it's `inmemory` cache. Unfortunately it's not expiring and boundless. We need to turn it off.